nvpair-sys/src/bindings.rs

```diff
@@ -493,95 +493,95 @@ extern "C" {
     pub fn nvlist_add_boolean_array(
         arg1: *mut nvlist_t,
         arg2: *const ::std::os::raw::c_char,
-        arg3: *mut boolean_t::Type,
+        arg3: *const boolean_t::Type,
```

Comment:
If we were to merge this upstream, presumably they would want nvpair-sys to continue working with old zfs installs, which don't have the `const` changes.  Would it be a problem that the bindings don't exactly match the C code/headers, or do we get away with it because the "const" doesn't actually change the C API (e.g. libnvpair.so still has the same assembly, we're just declaring it more precisely)

Reply:
I believe that because the const changes aren't actually changing the API, they're just properly documenting it, it should be fine. In general, however, this is probably the responsibility of the packaging layer. If we added a new libnvpair function and wanted to use it in nvpair-sys, whatever's creating the actual packages that the end users install should specify that nvpair-sys v.1.2.3 depends on libnvpair v4.5.6. I don't think there's a way to express that version dependency in rust or cargo itself, but I could be missing something.